code/modules/vehicles/mecha/mecha_defense.dm

```diff
@@ -0,0 +1,324 @@
+		return
+	if(construction_state == MECHA_LOOSE_BOLTS)
+		construction_state = MECHA_SECURE_BOLTS
+		to_chat(user, span_notice("You tighten the securing bolts."))
+
+/obj/vehicle/sealed/mecha/crowbar_act(mob/living/user, obj/item/I)
+	..()
+	. = TRUE
+	if(construction_state == MECHA_LOOSE_BOLTS)
+		construction_state = MECHA_OPEN_HATCH
+		to_chat(user, span_notice("You open the hatch to the power unit."))
+		return
+	if(construction_state == MECHA_OPEN_HATCH)
+		construction_state = MECHA_LOOSE_BOLTS
+		to_chat(user, span_notice("You close the hatch to the power unit."))
+
+/obj/vehicle/sealed/mecha/welder_act(mob/living/user, obj/item/W)
+	. = ..()
+	if(user.a_intent == INTENT_HARM)
+		return
+	. = TRUE
+	if(obj_integrity < max_integrity)
+		if(!W.use_tool(src, user, 0, volume=50, amount=1))
+			return
+		user.visible_message(span_notice("[user] repairs some damage to [name]."), span_notice("You repair some damage to [src]."))
+		obj_integrity += min(10, max_integrity-obj_integrity)
+		if(obj_integrity == max_integrity)
+			to_chat(user, span_notice("It looks to be fully repaired now."))
+		return
+	to_chat(user, span_warning("[src] is at full integrity!"))
+
+/obj/vehicle/sealed/mecha/proc/full_repair(charge_cell)
+	obj_integrity = max_integrity
+	if(cell && charge_cell)
+		cell.charge = cell.maxcharge
+	if(internal_damage & MECHA_INT_FIRE)
+		clear_internal_damage(MECHA_INT_FIRE)
+	if(internal_damage & MECHA_INT_TEMP_CONTROL)
+		clear_internal_damage(MECHA_INT_TEMP_CONTROL)
+	if(internal_damage & MECHA_INT_SHORT_CIRCUIT)
+		clear_internal_damage(MECHA_INT_SHORT_CIRCUIT)
+	if(internal_damage & MECHA_INT_TANK_BREACH)
+		clear_internal_damage(MECHA_INT_TANK_BREACH)
+	if(internal_damage & MECHA_INT_CONTROL_LOST)
+		clear_internal_damage(MECHA_INT_CONTROL_LOST)
+
+/obj/vehicle/sealed/mecha/do_attack_animation(atom/A, visual_effect_icon, obj/item/used_item, no_effect)
+	if(!no_effect && !visual_effect_icon)
+		visual_effect_icon = ATTACK_EFFECT_SMASH
+		if(damtype == BURN)
+			visual_effect_icon = ATTACK_EFFECT_MECHFIRE
+		else if(damtype == TOX)
+			visual_effect_icon = ATTACK_EFFECT_MECHTOXIN
+	..()
+
+/obj/vehicle/sealed/mecha/obj_destruction(damage_amount, damage_type, damage_flag)
+	if(wreckage)
+		var/mob/living/silicon/ai/AI
+		for(var/crew in occupants)
+			if(isAI(crew))
+				if(AI)
+					var/mob/living/silicon/ai/unlucky_ais = crew
+					unlucky_ais.gib()
+					continue
+				AI = crew
+		var/obj/structure/mecha_wreckage/WR = new wreckage(loc, AI)
+		for(var/obj/item/mecha_parts/mecha_equipment/E in flat_equipment)
+			if(E.detachable && prob(30))
+				WR.crowbar_salvage += E
+				E.detach(WR) //detaches from src into WR
+				E.activated = TRUE
+			else
+				E.detach(loc)
+				qdel(E)
+		if(cell)
+			WR.crowbar_salvage += cell
+			cell.forceMove(WR)
+			cell.charge = rand(0, cell.charge)
+			cell = null
+	return ..()
+
+/obj/vehicle/sealed/mecha/proc/ammo_resupply(obj/item/mecha_ammo/A, mob/user,fail_chat_override = FALSE)
```

Comment:
rename A to ammo or something like that, it gets unclear

Reply:
done